### DISCUSSION TITLE
Fill canvas

### DISCUSSION THREAD

**Discussion by F-Burning** [opentoonz/opentoonz#6532] [Ideas]
Here's a quick demo:

https://user-images.githubusercontent.com/74098734/114232335-9e3cb700-9973-11eb-99d8-ed368a237915.mp4

To make it more "easy", this only would work in Raster and Toonz Raster level (preferible in Toonz raster)

PD: Apparently this is very difficult to achieve in Opentoonz, a real challenge, but if someone is able to achieve it, traditional artist will be eternally grateful 🙏

**Comment by oliver-dreamwalker** [comment]
@F-Burning , I think the option "Fill all level" should be called "Canvas" to keep it simple.

**Comment by gab3d** [comment]
@F-Burning 
I'm always in favor of having more options, so in this sense it'd be good to have this ability. 👍 

Besides that, I often find that most people asking for being able to fill drawings in another layer, are really trying to literally apply in OpenToonz workflows used in other programs, that OpenToonz resolves in its own unique (and clever) way.

So I'd like to ask you what are the benefits you expect to gain after you have the paint in a separate layer?

**Comment by artisteacher** [comment]
If I needed a fill on its own level for some reason, I would just clone the cells and fill on the original level as needed, using "Areas" as the mode. Afterwards I would set the opacity of the line style to 0 (for the original level). The cloned level would display the line work and the original level would just display fills.

Personally, I wouldn't use the term "Fill Canvas" - that implies something different. Perhaps "Fill Across Columns" or "Fill Through Layers".

**Comment by F-Burning** [comment]
@gab3d  @artisteacher 

Although that is the example I have given, it is not the true purpose, what happens is that I have simplified the idea so that it is better understood.


https://user-images.githubusercontent.com/74098734/114262918-72085100-99da-11eb-9da3-b2905946c56a.mp4


Notice how the animation of the hair and clothes can be filled because the layer recognizes the line of the body (the other layer). I can even add the shadow of the hair on the skin of the face.


https://user-images.githubusercontent.com/74098734/114263678-f957c380-99de-11eb-8ed3-2ae240fe8844.mp4



This has been used every day in the anime industry for 20 years, and it helps streamline work times as well as maintain a cleaner workflow by not needing extra layers. It can be very useful for secondary animation.


I think that if Opentoonz has tools or options oriented to anime industry, such as autopaint lines, it should go a step further, because the feeling it leaves you is that the software itself limits your possibilities, and some small additions like this would be needed to be taken as a serious option for this purposes.

**Comment by gab3d** [comment]
@F-Burning 
ok, now it's better understood what the technique you intend to apply is, and the benefits obtained from it.
it seems legit. 👍 

I suppose it'd be easier to implement it for Raster levels, but I agree it'd take better advantage of the possibilities of OT if it is possible to implement it for Toonz Raster levels.
The situation with Toonz Raster is that it requires to have closed "Lines" to be able to define fill-able "Areas", so it should require the Lines of the other visible levels to be copied onto the current one (behind the pre-existent Lines there), prior to the fill operation, to be able to accomplish it.
I think the said Lines could be copied using a newly created black color with alpha=0 to allow for the filling operation but don't interfere visually with the intended result or the lines already drawn in the current level.

I'm not implying it can or can't be done, but just trying to think out what it'd take to implement it.

**Comment by F-Burning** [comment]
It's a good idea @gab3d 👍 

Another option (if finally this is not possible) could be allow this feature in Raster Levels, that works such a bitmap, and then an option to convert Raster to Toonz Raster in the level panel.

Would you like it?

Edit: @artisteacher remember me that this not would work because Toonz raster use Lines and Areas 😅

**Comment by RodneyBaker** [comment]
I don't quite follow the request here but...

I am curious how closely Tahoma2D's Reference Fill feature meets the need of this request.
@F-Burning ?
